src/main/scala/is/hail/sparkextras/OrderedRDD2.scala

```diff
@@ -121,7 +121,9 @@ object PartitionKeyInfo2 {
     val rng = new java.util.Random(seed)
     val samples = new Array[WritableRegionValue](sampleSize)
 
-    var i = 0
+    samples(0) = WritableRegionValue(typ.pkType, f0)
```

Comment:
Good catch!  You need to check sampleSize > 0 here.

Reply:
No, `sampleSize == 0` should be allowed.  My code below supports
 it.  I think you just need:  `if (0 < sampleSize) samples(0) = ...`.